gms/gossiper.cc

```diff
@@ -762,16 +762,28 @@ future<> gossiper::failure_detector_loop_for_node(gms::inet_address node, int64_
     auto diff = gossiper::clk::duration(0);
     auto echo_interval = std::chrono::milliseconds(2000);
     auto max_duration = echo_interval + std::chrono::milliseconds(_failure_detector_timeout_ms());
+    abort_source as;
+    auto tmr = timer([&as] {
+        as.request_abort_ex(timed_out_error());
+    });
+    auto sub = _abort_source.subscribe([&] () noexcept {
+        tmr.cancel();
+        as.request_abort_ex(abort_requested_exception());
+    });
     while (is_enabled()) {
```

Comment:
Can't we use _abort_source directly?

Reply:
I think I see a reason why not, but the code doesn't consider it. If the timer ever fires, `as` becomes poisoned. We need a new one every iteration.

Reply:
Right again.

Reply:
Hmm, theoretically gossip can be stopped gracefully without requesting abort.
But I'll add a check like in https://github.com/scylladb/scylladb/blob/563998b69a1cb2db4e4dba7eb9f9703204588ffd/gms/gossiper.cc#L809